src/models/piranha_arguments.rs

```diff
@@ -331,20 +331,13 @@ mod piranha_arguments_test;
 impl SourceCodeUnit {
   /// Delete the comment associated to the deleted code element
   pub(crate) fn _delete_associated_comment(
-    &mut self, edit: &Edit, applied_edit: &mut InputEdit, parser: &mut tree_sitter::Parser,
-  ) {
-    // Check if the edit kind is "DELETE something"
-    if *self.piranha_arguments().cleanup_comments() && edit.replacement_string().is_empty() {
-      let deleted_at = edit.p_match().range().start_point.row;
-      if let Some(comment_range) = self._get_comment_at_line(
-        deleted_at,
-        *self.piranha_arguments().cleanup_comments_buffer(),
-        edit.p_match().range().start_byte,
-      ) {
-        debug!("Deleting an associated comment");
-        *applied_edit = self._apply_edit(&Edit::delete_range(self.code(), comment_range), parser);
-      }
+    &mut self, edit: &Edit, parser: &mut tree_sitter::Parser,
```

Comment:
from `apply_edit` we call `_delete_associated_comment` and from `_delete_associated_comment` we recursively call `apply_edit`. This ensures that can delete a series of comments.

Let's say `apply_edit` deletes `int foo` in the below example.
```
// some 
// comment 
int foo;
```

becomes 

```
// some 
// comment 
```

now `_delete_associated_comment` is triggered, and now we produce
```
// some 
```

Since `_delete_associated_comment` is effectively called recursively 

```
<all comments are deleted>
```

Reply:
Can we have this documented in a comment in the code in addition to in the PR review comments? Possibly in the documentation for this function.